### DISCUSSION TITLE
How to contribute to the werf project

### DISCUSSION THREAD

**Discussion by mehrdadbn9** [werf/werf#6209] [General]
Hi every one i wanna know how should contribute and most needed one at community 
 please tell the one people that are not familiar as well

**Comment by shurup** [comment]
Hi @mehrdadbn9, and thank you for your question! Let me quote my recent answer to a similar question in [our Telegram chat](https://t.me/werf_io).

werf itself and its ecosystem (including Nelm, trdl, kubedog, and other projects) are implemented in Go, meaning it’s a must-have requirement to bring code changes. There are a few [**good first issues**](https://github.com/werf/werf/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) where you can see if something fits your expectations.

Currently, we are actively working on the [**Nelm**](https://github.com/werf/nelm) project. In the following weeks, it will be released as a standalone tool, and the community's help with trying & improving it will be very much appreciated. You'll see a related announcement here in Discussions when it's out.

At the same time, there is always room for improvement in other areas, such as docs. Here is where your contributions to werf are especially welcome at the moment:

1. **Documentation / guides**. We have Kubernetes guides aimed at developers who want to learn about K8s and practice their new knowledge using werf. They are massive and have ready-to-use editions for different programming languages and web frameworks, such as JavaScript/Node.js, Python/Django, Ruby/Rails, etc. You can find them online [here](https://werf.io/guides.html) ­and, after trying them, suggest any improvements using the werf/website repo [here](https://github.com/werf/website/tree/main/pages_en/guides). Please note that the content there relies heavily on the inclusion mechanism provided by Jekyll, so lots of reused documents will be found in [another directory](https://github.com/werf/website/tree/main/_includes/en/guides). 

2. **Documentation / configurator**. Not long ago, we launched a brand new section of documentation guiding users to install & run werf following various use cases, platforms, CI systems, etc. You can view it [here](https://werf.io/getting_started/) and find/improve its main content in the werf/website repo [here](https://github.com/werf/website/tree/main/bin/configurator/static/_includes/en/configurator). 

3. **Documentation / other pages**. Actually, it’s always good to have another couple of eyes on all documentation pages. Most of its content can be found [here](https://github.com/werf/werf/tree/main/docs/pages_en). 

4. **Custom Resources tracking rules**. If you want to use werf to deploy some specific Kubernetes resources which are not auto-detected by werf’s deployment tracker, you’re welcome to add them so other users might benefit. Find more details in this discussion: #5888

5. **Tech articles**. While it’s not a direct contribution to the werf repos, all third-party articles about werf are a HUGE help for the project. It can be a general overview of werf, a manual on installing and starting using werf, a more detailed explanation of any werf’s features, etc. You’re most welcome to post it on the platform of your choice (Medium, dev.to, FAUN.dev, LinkedIn, etc.)! We are happy to: a) assist with fact-checking all the information and b) promote your article via our resources (making a post for X/Twitter from the official werf account & adding it to the [publications page](https://werf.io/publications.html)). By the way, if a formal GitHub contribution is essential for you, you’ll be able to make your PR by adding such a link to our website, too.

Please, feel free to ask any questions you might have.
